docs/openapi/howto/archboard_feedback.md

```diff
@@ -0,0 +1,152 @@
+- [Need two or more service clients (regardless of if some needs subclients or not)](#need-two-or-more-service-clients-regardless-of-if-some-needs-subclients-or-not)
+- [Rename a model name](#rename-a-model-name)
+- [Rename a model attribute](#rename-a-model-attribute)
+- [Operation should be pageable](#operation-should-be-pageable)
+
+## Need for more/less SDK package(s)
+
+If you got as a feedback that your API needs to split across several packages, or two packages should be merged into one, then
+structure your Swaggers to have exactly one folder per package. Most of the work is re-organizing files and Readme. Example of hierarchy on 
+the [RestAPI spec repo](https://github.com/Azure/azure-rest-api-specs):
+
+```
+specification/
+├─ myservice/
+│  ├─ data-plane/
+│  │  ├─ package1/
+│  │  │  ├─ Microsoft.MyService/ (swaggers for package 1)
+│  │  │  ├─ readme.md
+│  │  ├─ package2/
+│  │  │  ├─ Microsoft.MyService/ (swaggers for package 2)
+│  │  │  ├─ readme.md
+```
+
+## Rename an operation
+
+Operation name are directly OperationID in Swagger. Just rename the OperationID directly.
+
+Note: If your operation contains an `_`, just change the second part as the first is an operation group.
+
+``` json
+ "operationId": "SomethingToDoOperation",
+ ...
+ "operationId": "MyGroup_SomethingElseToDoOperation",
+```
+becomes
+``` json
+ "operationId": "BuildTheThing",
+ ...
+ "operationId": "MyGroup_BuildSomethingElse",
+```
+
+## Rename an operation group (or rename a sub-client)
+
+Operation groups are the first part of an OperationID in Swagger, before the `_` character. To rename an operation group, you need to rename all OperationID that
+starts with the operation group prefix (there is likely many of them).
+
+``` json
+ "operationId": "MyGroup_BuildSomethingElse",
+```
+becomes
+``` json
+ "operationId": "NewGroup_BuildSomethingElse",
+```
+
+## Need one client to have all operations directly on it (no sub-client)
+
+Don't use operation groups in your OperationID (remove them if they exist). For any operation with an `_` character, remove the first part.
+
+``` json
+ "operationId": "MyGroup_BuildSomethingElse",
+```
+becomes
+``` json
+ "operationId": "BuildSomethingElse",
+```
+
+## Need one client/builder with sub-clients
+
+-	Use operation groups, and name them as you want your subclients to be called. Operation group can be create in Swagger by using the `_` character in the OperationID 
+(`OperationGroupName_OperationName`)
+-	For C#, use `single-top-level-client: true`
+
+For instance, this:
+``` json
+ "operationId": "MyGroup_BuildSomethingElse",
+```
+
+Will create a subclient for MyGroup, that will have an operation called BuildSomethingElse.
+In various languages this could looks like (pseudo-code):
+```
+MyServiceClient(endpoint, credentials).MyGroup().BuildSomethingElse() // C#
+
+MyServiceClient(endpoint, credentials).my_group.build_something_else() # Python
+```
```

Comment:
There is currently no support on the sub-clients in Java design guide https://azure.github.io/azure-sdk/java_introduction.html (they were in codegen long time ago, but explicitly forbidden by Java guideline).

Therefore, 1 operation group directly map to 1 client/builder pair.

If as next section, another service client is required, lib just get more client/builder (which may take a different options e.g. different key or token config, different host etc.).
It is a flat hierarchy.

@srnagar Please correct me if I am wrong.

Reply:
I discussed with @srnagar actually before writing this, that Java by default with have one central builder, that can build sub-clients. That check the box in my book: there is on entry point (even if it's not an actual client).

Reply:
OK, this makes more sense as we had have this as default more recently (till 3 months ago).

Today's default is multiple builder (instead of central builder).
Dev can switch to older behavior by having `generate-builder-per-client=false` flag.

Feature request on multiple builder https://github.com/Azure/autorest.java/issues/1328

Reply:
Ok, so I need to add "generate-builder-per-client=false" to this guideline here?

Reply:
Yes, dev can set `generate-builder-per-client: false` in readme.java.md to enable this.